src/messaging/ExchangeContext.h

```diff
@@ -115,6 +115,18 @@ class DLL_EXPORT ExchangeContext : public ReliableMessageContext, public Referen
      */
     void WillSendMessage() { mFlags.Set(Flags::kFlagWillSendMessage); }
 
+    /**
+     * A notification that we should keep waiting for a response.  The usual way
+     * this would be used is if we notify our consumer that a response timed out
+     * and they want to keep waiting (via OnResponseTimeout).
+     *
+     * If this function returns error, the exchange was not able to start
+     * waiting for a response and the consumer should decide how it wants to
+     * handle that.  Closing the exchange is not necessary if the call was made
+     * under OnResponseTimeout(); the exchange will handle this itself.
+     */
+    CHIP_ERROR KeepWaitingForResponse();
```

Comment:
This seems additional feature which allow delegate re-start the response timer, I was wondering is there any real use case for this function? If the app want to extend the response timeout, why not just use a bigger timeout value?

Reply:
I could drop this and just have the semantics be "timeout means the exchange closes", sure.  Would you prefer that @yufengwangca ?  I was trying to preserve the existing semantic possibilities, which allow the consumer to keep waiting even after timeout.  But I agree we have nothing that does that right now, so could just drop it for the moment.

Reply:
Yes, I prefer to drop this to make API simple until we do have use case for this feature.

Reply:
OK, coming right up.